plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -85,24 +85,25 @@ public void loadConnectors(Settings settings) {
   }
 
   @Override
-  public StorageEngine getStorageEngine(String catalog) {
-    if (!storageEngineMap.containsKey(catalog)) {
-      return defaultOpenSearchStorageEngine;
-    }
-    return storageEngineMap.get(catalog);
+  public Set<Catalog> getCatalogs() {
+    return new HashSet<>(catalogMap.values());
   }
 
   @Override
-  public Set<String> getCatalogs() {
-    return Collections.unmodifiableSet(storageEngineMap.keySet());
+  public Catalog getCatalog(String catalogName) {
+    if (!catalogMap.containsKey(catalogName)) {
```

Comment:
should it throw exception? i think the logic already guarantee all the catalog name is resolved by rule. https://github.com/opensearch-project/sql/blob/2.x/core/src/main/java/org/opensearch/sql/analysis/model/CatalogName.java#L17

Reply:
`.opensearch` is a placeholder name which I want to don't include. Also when we query SHOW CATALOGS I am not providing `.opensearch` in output. 

So I am not maintaining `.opensearch` catalog in catalogMap. What do you suggest? Should I treat `.opensearch` as another catalog. That also seems reasonable.